js/exporter/jspdf/export.js

```diff
@@ -125,24 +121,25 @@ export const Export = {
         }
     },
 
-    setColumnWidths: function(autoTableOptions, columns, keepColumnWidths) {
-        const columnStyles = autoTableOptions.columnStyles;
+    getDefaultPdfColumnWidths(autoTableWidth, columnWidths) {
+        if(isNumeric(autoTableWidth)) {
+            const tableWidth = columnWidths.reduce((a, b) => { return a + b; });
+            return columnWidths.map((columnWidth) => {
+                return autoTableWidth * columnWidth / tableWidth;
+            });
+        }
+    },
 
-        for(let i = 0; i < columns.length; i++) {
+    updateColumnWidths: function(jsPDFDocument, autoTableOptions, pdfColumnWidths, keepColumnWidths) {
```

Comment:
rename to 'setColumnWidthsIfEmpty: function(jsPDFDocument, autoTableOptions, pdfDefaultColumnWidths)' ?

Reply:
I understand your idea - set width only for columns that have no width.
There is a problem. For example:
`// dxDataGrid.width === 220; columnWidths = [105, 115];`
Customer settings: `{ tableWidth: 400, columnStyles: { 0: { cellWidth: 100 } } }`
`// columnWidths = [191, 209]; // this.getDefaultPdfColumnWidths(...);`
In our logic will be configure width of cell for second column: 400 * 115 / 220 = 209;
In summary will be:
`{ tableWidth: 400, columnStyles: { 0: { cellWidth: 100 }, 1: { cellWidth: 209 } } }`
`400 !== 100 + 209`, like [this](https://user-images.githubusercontent.com/16776723/91630811-54532200-e9dd-11ea-91c5-5850ff33fc4f.png)

IMHO:
We need to configure the width for all cells in case when 'keepColumnWidths === true'
if the user wants to customize widths, then he needs set `keepColumnWidths` to false.

Any idea?

Reply:
ok, setColumnWidths, no conditions